### PR TITLE
Spanish version

### PR DESCRIPTION
**Pull request by ericho**
A translation from English to Spanish of the course.

### CODE REVIEW

**Comment by skade** [comment]
Oh, wait, that means I should really put the english translation on the back burner... thanks!

**Comment by ericho** [comment]
I can also learn some german in the meanwhile :smile:

File: presentation/chapters/es-ES/data-structures.chapter [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+fn main() {
+    let direction = Direction::Left;
+}
+```
+
+Las diferentes elecciones de un Enum son llamadas "variantes".
+
+---
+
+## Enums con valores
```
Comment (emig) [inline presentation/chapters/es-ES/data-structures.chapter]: Enums con valores como variantes ?

File: presentation/chapters/es-ES/documentation.chapter [hunk 2]
```diff
@@ -0,0 +1,96 @@
+# Documentación
+
+---
+
+## `rustdoc`
+
+Rust provee una herramienta de documentación estándar llamada `rustdoc`. Es comúnmente usada a través de `cargo doc`.
+
+Debido a esto el código de Rust es casi siempre documentado con un formato común.
```
Comment (emig) [inline presentation/chapters/es-ES/documentation.chapter]: es --> está    ah, ser or estar, metafísica pura

**Comment by skade** [comment]
In the meantime, you can also add spanish as a language here, and maybe create a course summary in `courses` (that's basically the entry point into the languages)

https://github.com/skade/rust-three-days-course/blob/master/presentation/index.html#L43

File: presentation/chapters/es-ES/documentation.chapter (hunk 2 above)
Comment (ericho) [inline presentation/chapters/es-ES/documentation.chapter]: Gracias ! Se me paso eso.

File: presentation/chapters/es-ES/data-structures.chapter (hunk 1 above)
Comment (ericho) [inline presentation/chapters/es-ES/data-structures.chapter]: Thanks, that's better 😄

File: presentation/chapters/es-ES/working-with-nightly.chapter [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+```rust
+#![feature(asm, no_std)]
+```
+
+---
+
+## Plugins de compilador
+
+Los plugins del compilador agregar características adicionales a Rust. Por ejemplo:
```
Comment (emig) [inline presentation/chapters/es-ES/working-with-nightly.chapter]: agregar --> agregan

File: presentation/chapters/es-ES/error-handling.chapter [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        println!("It didn't work!")
+    }
+}
+```
+
+---
+
+## Usando `Result` con `?`
+
+Use `?` en funciones con múltiples posibles fallos.
```
Comment (emig) [inline presentation/chapters/es-ES/error-handling.chapter]: múltiples fallos posibles.   

me suena mejor..

File: presentation/chapters/es-ES/error-handling.chapter [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+<pre class="diagram">
++-----+   +-----+   +-----+
+| T,E |-->| U,E |-->| U,X |
++-----+   +-----+   +-----+
+</pre>
+
+---
+
+## Mapando valores `Result`
```
Comment (emig) [inline presentation/chapters/es-ES/error-handling.chapter]: Mapando --> Mapeando

File: presentation/chapters/es-ES/generics-basics.chapter [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    Err(E)
+}
+
+fn main() {
+    let file = std::fs::File::open("I don't exist!");
+    println!("{:?}", file);
+}
+```
+
+Describe cuando una operación fue exitosa y devuelve ya sea el valor un error.
```
Comment (emig) [inline presentation/chapters/es-ES/generics-basics.chapter]: ya sea el valor un error.  -> ya sea el valor `o` un error

File: presentation/chapters/es-ES/installation.chapter [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+# Primeros pasos
+
+Ejecute los siguientes comandos:
+
+```sh
+$ rustup component add rust-src
+$ rustup component add rust-docs
+```
+
+Esto carga el codigo fuente y documentación de la librería estándar
```
Comment (emig) [inline presentation/chapters/es-ES/installation.chapter]: codigo -> código

File: presentation/chapters/es-ES/iterators.chapter [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+        println!("{}", item);
+    }
+}
+```
+
+---
+
+## `filter()`
+
+Filtra valores no deseados, saltando calculos adicionales en ellos:
```
Comment (emig) [inline presentation/chapters/es-ES/iterators.chapter]: cálculos

File: presentation/chapters/es-ES/standard-types.chapter [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+
+Hay distintos tipos en Rust.
+
+Impulsan el poderoso sistema de tipos para conseguir tareas fundamentales.
+
+---
+
+## Visión general
+
+* `Option<T>` - Remueve la necesidad de la primitivia `null`.
```
Comment (emig) [inline presentation/chapters/es-ES/standard-types.chapter]: primitiva

File: presentation/chapters/es-ES/strings.chapter [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+Hay distintos tipos de cadenas en Rust.
+
+Los más comunes son `String` y `&str`.
+
+---
+
+## `String`
+
+* *Posee* el dato que almacena, y puede mutar libremente.
+* Existe como un puntero a algunos bytes, a longitu y capacidad.
```
Comment (emig) [inline presentation/chapters/es-ES/strings.chapter]: longitud

File: presentation/chapters/es-ES/testing.chapter [hunk 11]
```diff
@@ -0,0 +1,208 @@
...
+
+Probar es fundamental en Rust.
+
+Pruebas de unidad, integración y documentación vienen integradas.
+
+---
+
+## Organizando pruebas
+
+Las pruebas típicamente terminan en una de 4 posibles lugares:
```
Comment (emig) [inline presentation/chapters/es-ES/testing.chapter]: uno de estos 4 posibles lugares

File: presentation/chapters/es-ES/testing.chapter [hunk 12]
```diff
@@ -0,0 +1,208 @@
...
+
+## Submodulo `tests`
+
+* Permite probar si la funcionalidad puede ser usada en otro punto del proyecto.
+* Para probar APIs privadas y funcionalidad.
+* Buenas para probar procesos esperados y casos de uso.
+
+---
+
+## Submodulo `tests`
```
Comment (emig) [inline presentation/chapters/es-ES/testing.chapter]: Submódulo

File: presentation/chapters/es-ES/testing.chapter (hunk 11 above)
Comment (emig) [inline presentation/chapters/es-ES/testing.chapter]: Quizás se puede usar `test`  en lugar de <prueba>

File: presentation/chapters/es-ES/traits.chapter [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+
+---
+
+## Borrowing y Ownership de `self`
+
+No es diferente de lo normal, pero al principio un poco desconocido.
+
+* El préstamo en una función _simultáneamente cede self_.
+* Esto es especialmente aplicable para préstamos mutables.
+* `self` sin ` & ` devuelve la propiedad al valor del contexto llamante.
```
Comment (emig) [inline presentation/chapters/es-ES/traits.chapter]: que lo invoca

**Comment by skade** [comment]
Hey, just to give you a clear indication where things are heading:

* I'm currently writing a script to pull out all code samples from the translations
* Then, I'll merge the current state of this pull request
* Then move on to german -> english translations.

Sadly, I was a bit sick yesterday, so all things are delayed, but everything is on track.

Thanks again for this PR already 👍

**Comment by ericho** [comment]
Hey, no worries, I hope you feel better :)

**Comment by skade** [comment]
So, with 558c6b9 in, let's move on to the merging :).

**Comment by skade** [comment]
Sooo, I also added `courses/spanish.md`, which links to the spanish set of chapters. As I don't speak Spanish, it's currently German, but shouldn't be too hard to change.

What it gives you is a nice table of contents, viewable [here](https://skade.github.io/rust-three-days-course/courses/spanish.html).

**Comment by ericho** [comment]
:O great! I'll update that file.

**Pull request merged**